packages/ag-charts-community/src/chart/series/cartesian/barSeries.ts

```diff
@@ -705,10 +719,9 @@ export class BarSeries extends AbstractBarSeries<
         const direction = this.getBarDirection();
 
         opts.datumSelection.each((rect, datum) => {
-            const style = this.getItemStyle(datum, opts.isHighlight);
-            applyShapeStyle(rect, style, fillBBox);
+            applyShapeStyle(rect, datum.style, fillBBox);
 
-            const cornerRadius = style.cornerRadius ?? 0;
+            const cornerRadius = datum.style.cornerRadius ?? 0;
             rect.topLeftCornerRadius = datum.topLeftCornerRadius ? cornerRadius : 0;
             rect.topRightCornerRadius = datum.topRightCornerRadius ? cornerRadius : 0;
             rect.bottomRightCornerRadius = datum.bottomRightCornerRadius ? cornerRadius : 0;
```

Comment:
Maybe `getItemStyle` should resolve each corner radius, then these functions could just be `rect.setProperties(datum.style)`. Not urgent though